### PR TITLE
Enable security compile and link time features

### PR DESCRIPTION
**Pull request by jvaltin**
This change set enables golang compiler and linking security features.

### CODE REVIEW

**Comment by jvaltin** [comment]
This change set should make memory corruption exploitation more of a challenge.

Related reading: 
https://rain-1.github.io/golang-aslr.html
https://developers.redhat.com/blog/2018/03/21/compiler-and-linker-flags-gcc/

File: frontends/qt/server/Makefile [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
+CFLAGS += ${SECURITYFLAGS}
+LFLAGS += ${EXTLDFLAGS}
+export CXXFLAGS
+export CFLAGS
+export LFLAGS
+export CGO_LDFLAGS_ALLOW=".*"
+export CGO_CFLAGS_ALLOW=".*"
 build:
-	CGO_ENABLED=1 go build -ldflags="-s -w" -buildmode=c-shared -o libserver.so
+	CGO_ENABLED=1 go build -buildmode=c-shared -ldflags -extldflags=${EXTLDFLAGS} -o libserver.so
```
Comment (benma) [inline frontends/qt/server/Makefile]: This does not work on macOS unfortunately: `running clang failed: exit status 1: unknown option -z`.

Please fix by making separate targets for now. We can fix if for macOS later.

Comment (jvaltin) [reply frontends/qt/server/Makefile]: Could you share the version of OS X, clang, ld, and golang? If you edit the EXTLDFLAGS, what works?

**Comment by benma** [comment]
macOS 10.13.6
```clang --version

Apple LLVM version 9.1.0 (clang-902.0.39.2)
Target: x86_64-apple-darwin17.7.0
Thread model: posix
```

Go 1.10.2

`ld` has no option to show the version. 

Will debug more tomorrow.

**Comment by jvaltin** [comment]
For OS X and for Windows, we will set different compiler flags. I'll push a branch with updates that cover these different build environments.

File: frontends/qt/server/server.go [hunk 2]
```diff
@@ -35,6 +35,7 @@ typedef struct ConnectionData {
 } ConnectionData;
 #endif
 */
+// #cgo CFLAGS: -O2 -D_FORTIFY_SOURCE=2 -fstack-check -fstack-protector-all -fPIC -fPIE
```
Comment (benma) [inline frontends/qt/server/server.go]: Same flags are in the server/Makefile. Are both needed?

**Comment by jvaltin** [comment]
I have a new branch which merges #11 and #13 into one set of changes.

**Pull request closed without merging**